src/mlpack/methods/ann/layer/convolution.hpp

```diff
@@ -162,6 +162,46 @@ class Convolution
   //! Modify the output height.
   size_t& OutputHeight() { return outputHeight; }
 
+  //! Get the input size
+  size_t const& InputSize() const { return inSize; }
+  //! Modify the input size.
+  size_t& InputSize() { return inSize; }
+
+  //! Get the output size.
+  size_t const& OutputSize() const { return outSize; }
+  //! Modify the output size.
+  size_t& OutputSize() { return outSize; }
+
+  //! Get the kernel height.
+  size_t const& KernelHeight() const { return kH; }
+  //! Modify the kernel height.
+  size_t& KernelHeight() { return kH; }
+
+  //! Get the kernel width.
+  size_t const& KernelWidth() const { return kW; }
```

Comment:
I was wondering if we should provide an interface that returns a tuple, so we could write ``auto [width, height] = KernelSize``. Let  me know what you think.

Reply:
Yes this did cross my mind, but I thought tuple was not probably there in c++11 but I checked the docs right now and well, it was there back then. I agree with you that tuple would a better choice. So I'll change it to a tuple representation wherever applicable. 

Moreover, Torch also uses tuples to pass these dimensions :)